### ISSUE TITLE
[Faceted Filters] Updated product count is not announced

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#115]
## Issue summary

When selecting a filter, the product count is updated visually. There is no notification of a successful facet selection. Users of assistive technology are not made aware of the content updates. This may lead to a confusing user experience for users of assistive technology.

<details>
  <summary>Screenshots</summary>

![Gif of faceted filters section shows Brand filter facet selection being made. Then, an arrow and shrug emoji point at the updated product count text.](https://user-images.githubusercontent.com/1392632/124607650-0b110180-de3c-11eb-903c-94bfccc6f025.gif)

</details>

#### HTML

```html
<p class="collection-product-count light">Showing 74 of 74 products</p>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported browser.
2. With an active screen reader, activate a facet control and make a selection.

### Expected

- Updated Product Count text to be announced, alerting the user to a successful selection and updated status.

### Actual

- Product Count text is not announced.

## Recommendation

Add the `role` attribute to the `.collection-product-count` container, setting its value to `status`. This role value supplies an implicit `aria-live="polite"` and `aria-atomic="true`" in order to announce the newly updated content after other announcements have been completed.

#### HTML

```html
<h2>
  <span class="collection-product-count" role="status">
    Showing 74 of 74 products
  </span>
</h2>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Robust
- **WCAG SC:** [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
- **Severity:** High
- **Effort:** Low
- **Related:** (#111, #116)

**Comment by svinkle** [comment]
@tyleralsbury Using [NerdeRegion](https://chrome.google.com/webstore/detail/nerderegion/lkcampbojgmgobcfinlkgkodlnlpjieb?utm_source=chrome-ntp-icon) I notice the `role="status"` element is dynamically removed and re-added to the DOM. The issue with this is macOS and Android screen readers announce the updated text, but Windows based screen readers _seemingly_ do not.

<details>
  <summary>Screenshot</summary>

![Screen shot of Dawn product listing page. NerdeRegion browser extension showing output for live announcement events, including the removal and addition of the live region DOM element.](https://user-images.githubusercontent.com/1392632/124973000-ff1e6e80-dff8-11eb-9d4b-3aedd54ce6bc.png)

</details>

Would it be possible to:
1. Leave the `role="status"` element in the DOM and only update the text node, or
2. Setup a new live region element that will be used for live updates only.

Screen readers are finicky and work best when the live region is in the DOM on page load and are never dynamically removed. This way they're able to "listen" to changes throughout the lifecycle of the page. (Also, multiple live regions create a major perf issue when the screen reader has to listen to more than one. Best to re-use the same element for updates.)

**Comment by tyleralsbury** [comment]
> Leave the role="status" element in the DOM and only update the text node

I think this is the best approach.  I'll see what I can do. Thanks, Scott.

**Comment by tyleralsbury** [comment]
I think that's what we did on the cart so I can probably use what I made for the cart for this as well.  Maybe I could even extract it into a helper function that both the cart and the filters can use.

**Comment by tyleralsbury** [comment]
Reopening the issue - this will probably be a bit bigger.  I'll keep the current implementation and wrap back around to this when I've got the chance.

**Comment by KaichenWang** [comment]
Due to update in design, the product count has been decoupled from the main-product grid. The updated implementation does not replace the container element (`role="status"`)  in the DOM; only its child `<p>` node is replaced. Tested on MacOS with Voiceover and Windows with NVDA and in both cases, the updated product count is announced following a facet selection.
